Daffodil 1856 - Updates to for Apache incubation

stevedlawrence opened the pull request:
This is split up into 4 commits to make things easier to review:

1. Remove old files
2. Move the directory layout. All changes are a 100% rename
3. Update package and xml namespaces to be apache based, adding backwards compatability for namespaces
4. Changes to add "Apache" and "incubating" where necessary, including updating the version to include "-incubating" and to be manually updated.

stevedlawrence commented on daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala:
```diff
@@ -1302,13 +1302,22 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
+
+      case (RefQName(_, "error", DAF_APACHE), args) => {
         SDW("Expression daf:error is deprecated. Use fn:error instead")
         DAFErrorExpr(functionQNameString, functionQName, args)
```
Changes to support xml namespace backwards compatibility.

stevedlawrence commented on daffodil-core/src/main/scala/org/apache/daffodil/dsom/IIBase.scala:
```diff
@@ -179,7 +179,17 @@ abstract class IIBase(xml: Node, xsdArg: XMLSchemaDocument, val seenBefore: IIMa
+        text
+      }
+    }
+  }
```
Supports backwards compatibility for schemas that still reference the old paths. For example, it is common to have an import with schemaLocation="edu/illinois/ncsa/daffodil/built-in-formats.xsd"

johnament commented:
Please note - the requirement is to have the word "incubating" in the source release.  There is no requirement that the version # has to include "incubating" but most people find this the easiest way to handle it.  It has the downside that your convenience binaries now unnecessarily include "-incubating" in them.

johnament commented on build.sbt:
```diff
@@ -21,7 +23,7 @@ testOptions in ThisBuild += Tests.Argument(TestFrameworks.JUnit, "-v")
 transitiveClassifiers := Seq("sources", "javadoc")
 
-resolvers in ThisBuild += "NCSA Sonatype Releases" at "https://opensource.ncsa.illinois.edu/nexus/content/repositories/releases"
+resolvers in ThisBuild += "NCSA Sonatype Releases" at "https://daffodil.apache.org/nexus/content/repositories/releases"
```
This url doesn't work.

stevedlawrence replied:
Actually, this line doesn't even belong in here anymore. Daffodil does not require dependencies from anything on NCSA. I'll remove this line.

johnament commented on daffodil-cli/DISCLAIMER:
```diff
@@ -0,0 +1,7 @@
+Apache Daffodil is an effort undergoing incubation at The Apache Software
```
The disclaimer should only have to appear once, in the source root.  If this is the root of the binary artifact that's fine, but its ideal if it can be a single entry.

stevedlawrence replied:
This was done because the daffodil-cli directory is a subproject, and there are some cases where SBT will prevent one from pulling in files outside of a subdirectory. This might not be one of those cases, I'll see if there is a way to only have one file.

stevedlawrence commented on daffodil-lib/src/main/scala/org/apache/daffodil/xml/XMLUtils.scala:
```diff
@@ -339,20 +339,24 @@ object XMLUtils {
+  val EXT_NS_APACHE = NS(DAFFODIL_EXTENSION_NAMESPACE_APACHE.uri)
+
+  private val DAFFODIL_INTERNAL_NAMESPACE = NS(DAFFODIL_EXTENSIONS_NAMESPACE_ROOT_APACHE + ":int")
+  val INT_PREFIX= "dafint"
```
New XML Namespace defined.  Support for the NCSA namespace is still provided for backwards compatibility. The internal namespace does not maintain comapt, since it's internal only.

johnament commented on daffodil-cli/src/main/scala/org/apache/daffodil/debugger/CLIDebuggerRunner.scala:
```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2012-2015 Tresys Technology, LLC. All rights reserved.
```
We'll typically only change package name once the SGA is on file and we can remove copyright.

stevedlawrence replied:
Ok. So maybe we should hold off on the two commits that change the directory layout and package names/XML namespaces for now.

stevedlawrence commented:
Regarding the "-incubating" comment in the version, I was just following what other projects seemed to do. If it's only required in the source release, I'd probably prefer to not include it in the version. It added a little extra complexity and makes the version numbers a bit messy.

mbeckerle commented:
I'd really like to schedule this change specifically. I.e., coordinate so all outstanding work is committed, then this change occurs. Because I have outstanding changes that include removal and renaming of files, movement of code from class to class and file to file. 

Past experience has told me that major rename/relabel events need to happen without putting a huge burden on those with major sets of outstanding changes. 

In other words. I'd like 60 days notice before this happens.

stevedlawrence commented:
Seems reasonable. I'll close this pull request and create a new one that just adds the Apache incubator stuff and a few other minor things, plus addresses the comments in this thread.
